openmsupply/application-manager-web-app
# #741 create locked review when assignment is locked

Fixes #741 

### Back-end branch PR
[#741F](https://github.com/openmsupply/application-manager-server/pull/378)

### Changes
- Show modal when review has status LOCKED when trying to submit:

- When review was locked before started (`reviewAssignment.isLocked`) a new action (done on back-end) setup for ON_REVIEW_CREATED will change the status of this review to LOCKED, and in the front end we show like this:

Review page doesn't have "Submission area":
![Screen Shot 2021-05-28 at 12 22 50 PM](https://user-images.githubusercontent.com/16461988/119912354-70f1aa00-bfaf-11eb-8367-426cee4bade5.png)

On the Review home, there isn't a link to view the Review (probably we should show a **View** option now):
<img width="1139" alt="Screen Shot 2021-05-28 at 12 22 32 PM" src="https://user-images.githubusercontent.com/16461988/119912364-764ef480-bfaf-11eb-82e5-a02c3c835410.png">

### Test
~~- First you need to run `yarn database_init` on the back-end branch
- Login as **Consolidator2** and go to the review page for new Application serial 45678: http://localhost:3000/application/45678/review
- Open in an incognito tab, login with user **Reviewer2** and also open the review page: http://localhost:3000/application/45678/review
- Click `start` and then approve all or decline one to get a decision read to submit. Leave like this
- Go to the other instance (as Consolidator2) and submit the LOQ. This will lock Reviewer2 review.
- Go to Reviewer2 now and try to submit... Should show the modal~~
~~- PENDING: Noticed a problem with the current implementation to get the latest ReviewStatus, so need to work on a fix for that.~~ Won't fix now

Another test - you will need to restart the database....

- Login as Consolidator 2, and submit review of http://localhost:3000/application/45678/review as LOQ
- Login as Reviewer 2, start the review and check that it is already locked: No submission area available
